platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -39,68 +64,216 @@
     if (isInit) {
       return;
     }
 
     metricsPrefix = prefix;
 
     for (String reporter : reporters) {
-
       switch (reporter.toLowerCase()) {
         case "console":
-          initConsoleMetricsReporter(reportInterval);
+          initConsoleMetricsReporter(reportIntervalSec);
+          break;
+        case "logging":
+          initLoggingMetricsReporter(reportIntervalSec);
           break;
         case "prometheus":
-          initPrometheusExporter();
+          initPrometheusExporter(reportIntervalSec);
           break;
         default:
           LOGGER.warn("Cannot find metric reporter: {}", reporter);
       }
     }
+
+    // Add the service name and other given tags to the default tags list.
+    DEFAULT_TAGS.add(new ImmutableTag("app", serviceName));
+    tags.forEach((key, value) -> DEFAULT_TAGS.add(new ImmutableTag(key, value)));
+
+    // Register different metrics with the registry.
+    new ClassLoaderMetrics(DEFAULT_TAGS).bindTo(METER_REGISTRY);
+    new JvmGcMetrics(DEFAULT_TAGS).bindTo(METER_REGISTRY);
+    new ProcessorMetrics(DEFAULT_TAGS).bindTo(METER_REGISTRY);
+    new JvmThreadMetrics(DEFAULT_TAGS).bindTo(METER_REGISTRY);
+    new JvmMemoryMetrics(DEFAULT_TAGS).bindTo(METER_REGISTRY);
+
+    new ProcessMemoryMetrics().bindTo(METER_REGISTRY);
+    new ProcessThreadMetrics().bindTo(METER_REGISTRY);
+
     for (String key : DEFAULT_METRIC_SET.keySet()) {
       METRIC_REGISTRY
           .registerAll(String.format("%s.%s", metricsPrefix, key), DEFAULT_METRIC_SET.get(key));
     }
     isInit = true;
   }
 
+  /**
+   * This method is deprecated since we'll be removing the Dropwizard metrics support in
+   * future releases.
+   */
+  @Deprecated
   public static void register(String metricName, Metric metric) {
     final String fullMetricName = String.format("%s.%s", metricsPrefix, metricName);
     if (!METRIC_REGISTRY.getNames().contains(fullMetricName)) {
       METRIC_REGISTRY.register(fullMetricName, metric);
     }
   }
 
+  /**
+   * Registers a Counter with the given name with the service's metric registry and reports it
+   * periodically to the configured reporters. Apart from the given tags, the reporting service's
+   * default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_counters for more details on the Counter.
+   */
+  public static Counter registerCounter(String name, Map<String, String> tags) {
+    return METER_REGISTRY.counter(name, addDefaultTags(tags));
+  }
+
+  /**
+   * Registers a Timer (without histograms) with the given name with the service's metric registry
+   * and reports it periodically to the configured reporters. Apart from the given tags, the
+   * reporting service's default tags also will be reported with the metrics.
+   *
+   * See https://micrometer.io/docs/concepts#_timers for more details on the Timer.
+   */
+  public static Timer registerTimer(String name, Map<String, String> tags) {
```

Comment:
Not sure how heavy these return types are, would it be worth abstracting them so that the actual metric library isn't exposed?

Reply:
Yeah I thought about it. `Timer` is the most heavy in terms of resources but is also very rich in functionality. See https://micrometer.io/docs/concepts#_timers 
Adding an abstraction means limiting some functionality, unless we're adding all the functionality in our abstraction (which is too much). I think I like the richness of `Timer` functionality and services can use it in different use cases.
Given that multiple metric libraries can usually co-exist (like DW and Micrometer are present in this PR), we can add other one if there is a real good use case. It should be rare to change these libraries.